ports/physx/portfile.cmake

```
@@ -89,67 +10,155 @@ vcpkg_from_github(
+    list(APPEND OPTIONS "-DTARGET_BUILD_PLATFORM=linux")
+elseif(VCPKG_TARGET_IS_ANDROID)
+    list(APPEND OPTIONS "-DTARGET_BUILD_PLATFORM=android")
+else()
+    message(FATAL_ERROR "Unhandled or unsupported target platform.")
+endif()
 
-	file(GLOB DEBUG_BINS ${SOURCE_PATH}/physx/bin/${BUILD_PATH}/${DEBUG_CONFIGURATION}/*.dll)
-	file(INSTALL ${DEBUG_BINS} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+if(VCPKG_TARGET_IS_OSX OR VCPKG_TARGET_IS_IOS)
+    list(APPEND OPTIONS "-DNV_FORCE_64BIT_SUFFIX=ON" "-DNV_FORCE_32BIT_SUFFIX=OFF")
+endif()
 
-	vcpkg_copy_pdbs()
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    list(APPEND OPTIONS "-DPX_GENERATE_STATIC_LIBRARIES=OFF")
+else()
+    list(APPEND OPTIONS "-DPX_GENERATE_STATIC_LIBRARIES=ON")
+endif()
+
+if(VCPKG_CRT_LINKAGE STREQUAL "dynamic")
+    list(APPEND OPTIONS "-DNV_USE_STATIC_WINCRT=OFF")
+else()
+    list(APPEND OPTIONS "-DNV_USE_STATIC_WINCRT=ON")
 endif()
 
-file(
-	GLOB
-	RELEASE_LIBS
-	${SOURCE_PATH}/physx/bin/${BUILD_PATH}/${RELEASE_CONFIGURATION}/*.lib
-	${SOURCE_PATH}/physx/bin/${BUILD_PATH}/${RELEASE_CONFIGURATION}/*.pdb
-	${SOURCE_PATH}/physx/compiler/${COMPILER}/sdk_source_bin/${RELEASE_CONFIGURATION}/*.pdb
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+    list(APPEND OPTIONS "-DPX_OUTPUT_ARCH=arm")
+else()
+    list(APPEND OPTIONS "-DPX_OUTPUT_ARCH=x86")
+endif()
+
+# Replicate PhysX's CXX Flags here so we don't have to patch out /WX and -Wall
+if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
+    set(PHYSX_CXX_FLAGS "/fp:fast /GF /GS- /GR- /Oy")
+    if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+        set(PHYSX_CXX_FLAGS "${PHYSX_CXX_FLAGS} /arch:sse2")
+    endif()
+elseif(VCPKG_TARGET_IS_ANDROID AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
+    set(PHYSX_CXX_FLAGS "-std=c++11 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections -funwind-tables -fomit-frame-pointer -funswitch-loops -finline-limit=300 -fno-strict-aliasing -fstack-protector")
+elseif(VCPKG_TARGET_IS_ANDROID AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+    set(PHYSX_CXX_FLAGS "-std=c++11 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections")
+elseif(VCPKG_TARGET_IS_ANDROID AND VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+    set(PHYSX_CXX_FLAGS "-std=c++11 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections -fpack-struct=8 -malign-double")
+elseif(VCPKG_TARGET_IS_ANDROID AND VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    set(PHYSX_CXX_FLAGS "-std=c++11 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections -mstackrealign -msse3")
+elseif(VCPKG_TARGET_IS_IOS)
+    set(PHYSX_CXX_FLAGS "-std=c++11 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections -fstrict-aliasing -Wstrict-aliasing=2")
+else()
+    set(PHYSX_CXX_FLAGS "-std=c++11 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections -fno-strict-aliasing")
+endif()
+list(APPEND OPTIONS "-DPHYSX_CXX_FLAGS:INTERNAL=${PHYSX_CXX_FLAGS}")
```

```suggestion
list(APPEND OPTIONS "-DPHYSX_CXX_FLAGS:INTERNAL=${VCPKG_CXX_FLAGS}")
```

Would you be happy with something like this? It seems like the options set here should be part of the toolchain if any rather than being clobbered either here or in the upstream port.

Notably: If any of these disagree with values that come from the toolchain the resulting physx binaries might not link with anything else in the catalog.

---

I would be fine with that. My goal was to remove the /WX and -Werror from the cxx flags. :smile: